{"text": "2023-10-28 07:19:38 | INFO    | __main__:stash:129 | run - {'image': '20231026', 'role': 'EpicPlayer', 'headless': True, 'self_supervised': True}\n", "record": {"elapsed": {"repr": "0:00:01.146683", "seconds": 1.146683}, "exception": null, "extra": {"image": "20231026", "role": "EpicPlayer", "headless": true, "self_supervised": true}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 129, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:19:38.447593+08:00", "timestamp": 1698448778.447593}}}
{"text": "2023-10-28 07:19:48 | INFO    | __main__:claim_epic_games:95 | Try to flush cookie - {'task': 'claim_epic_games'}\n", "record": {"elapsed": {"repr": "0:00:10.903006", "seconds": 10.903006}, "exception": null, "extra": {"task": "claim_epic_games"}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "claim_epic_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 95, "message": "Try to flush cookie", "module": "claim", "name": "__main__", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:19:48.203916+08:00", "timestamp": 1698448788.203916}}}
{"text": "2023-10-28 07:19:51 | SUCCESS    | epic_games.agent:_login:203 | login - {'result': 'Successfully refreshed tokens'}\n", "record": {"elapsed": {"repr": "0:00:14.017629", "seconds": 14.017629}, "exception": null, "extra": {"result": "Successfully refreshed tokens"}, "file": {"name": "agent.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py"}, "function": "_login", "level": {"icon": "‚úÖ", "name": "SUCCESS", "no": 25}, "line": 203, "message": "login", "module": "agent", "name": "epic_games.agent", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:19:51.318539+08:00", "timestamp": 1698448791.318539}}}
{"text": "2023-10-28 07:20:02 | SUCCESS    | epic_games.agent:flush_token:224 | flush_token - {'path': PosixPath('/home/runner/work/baipiao/baipiao/epic/user_data_dir/epic-games@934175043/ctx_cookie.json')}\n", "record": {"elapsed": {"repr": "0:00:25.107440", "seconds": 25.10744}, "exception": null, "extra": {"path": "/home/runner/work/baipiao/baipiao/epic/user_data_dir/epic-games@934175043/ctx_cookie.json"}, "file": {"name": "agent.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py"}, "function": "flush_token", "level": {"icon": "‚úÖ", "name": "SUCCESS", "no": 25}, "line": 224, "message": "flush_token", "module": "agent", "name": "epic_games.agent", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:20:02.408350+08:00", "timestamp": 1698448802.40835}}}
{"text": "2023-10-28 07:20:02 | WARNING    | epic_games.agent:get_order_history:389 | Failed to get order history, cookie may have expired - {}\n", "record": {"elapsed": {"repr": "0:00:25.419010", "seconds": 25.41901}, "exception": null, "extra": {}, "file": {"name": "agent.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 389, "message": "Failed to get order history, cookie may have expired", "module": "agent", "name": "epic_games.agent", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:20:02.719920+08:00", "timestamp": 1698448802.71992}}}
{"text": "2023-10-28 07:20:03 | DEBUG    | __main__:create_tasks:64 | Put task - {'title': 'Tandem: A Tale of Shadows', 'url': 'https://store.epicgames.com/en-US/p/tandem-a-tale-of-shadows-c3f55e'}\n", "record": {"elapsed": {"repr": "0:00:25.841461", "seconds": 25.841461}, "exception": null, "extra": {"title": "Tandem: A Tale of Shadows", "url": "https://store.epicgames.com/en-US/p/tandem-a-tale-of-shadows-c3f55e"}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 64, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:20:03.142371+08:00", "timestamp": 1698448803.142371}}}
{"text": "2023-10-28 07:20:03 | DEBUG    | __main__:create_tasks:64 | Put task - {'title': 'The Evil Within 2', 'url': 'https://store.epicgames.com/en-US/p/the-evil-within-2'}\n", "record": {"elapsed": {"repr": "0:00:25.842157", "seconds": 25.842157}, "exception": null, "extra": {"title": "The Evil Within 2", "url": "https://store.epicgames.com/en-US/p/the-evil-within-2"}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 64, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:20:03.143067+08:00", "timestamp": 1698448803.143067}}}
{"text": "2023-10-28 07:20:03 | INFO    | epic_games.agent:claim_weekly_games:232 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/tandem-a-tale-of-shadows-c3f55e'}\n", "record": {"elapsed": {"repr": "0:00:25.842558", "seconds": 25.842558}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/tandem-a-tale-of-shadows-c3f55e"}, "file": {"name": "agent.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 232, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:20:03.143468+08:00", "timestamp": 1698448803.143468}}}
{"text": "2023-10-28 07:20:15 | ERROR    | __main__:run:155 | An error has been caught in function 'run', process 'MainProcess' (3578), thread 'MainThread' (140663859370880): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/baipiao/baipiao/epic/src/claim.py\", line 159, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/baipiao/baipiao/epic/src/claim.py\", line 155, in run\n    await agent.stash()\n  File \"/home/runner/work/baipiao/baipiao/epic/src/claim.py\", line 148, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/baipiao/baipiao/epic/src/claim.py\", line 120, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/home/runner/work/baipiao/baipiao/epic/src/epic_games/agent.py\", line 248, in claim_weekly_games\n    await expect(cta_btn).to_have_text(\"View In Cart\")\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 20080, in to_have_text\n    await self._impl_obj.to_have_text(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 455, in to_have_text\n    await self._expect_impl(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 70, in _expect_impl\n    raise AssertionError(f\"{out_message}\\nActual value: {actual} {log}\")\nAssertionError: Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\nwaiting for locator(\"xpath=//aside//button[@data-testid='add-to-cart-cta-button']\")\n", "record": {"elapsed": {"repr": "0:00:37.717215", "seconds": 37.717215}, "exception": {"type": "AssertionError", "value": "Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\nwaiting for locator(\"xpath=//aside//button[@data-testid='add-to-cart-cta-button']\")", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/baipiao/baipiao/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 155, "message": "An error has been caught in function 'run', process 'MainProcess' (3578), thread 'MainThread' (140663859370880):", "module": "claim", "name": "__main__", "process": {"id": 3578, "name": "MainProcess"}, "thread": {"id": 140663859370880, "name": "MainThread"}, "time": {"repr": "2023-10-28 07:20:15.018125+08:00", "timestamp": 1698448815.018125}}}
